### ISSUE TITLE
After updating to Meteor 1.4.1, I am getting this error - Exception in defer callback: Error: Meteor.userId can only be invoked in method calls. Use this.userId in publish functions.

### ISSUE THREAD

**Issue by rnarayan** [jagi/meteor-astronomy#502]
I have been getting following error since I upgraded my app to 1.4.1. The app works fine though.

I am not even using any publish functions. I get all my data through methods.

Exception in defer callback: Error: Meteor.userId can only be invoked in method calls. Use this.userId in publish functions.
I20160910-20:55:42.773(-5)?     at AccountsServer.userId (packages/accounts-base/accounts_server.js:82:13)
I20160910-20:55:42.774(-5)?     at Object.Meteor.userId (packages/accounts-base/accounts_common.js:246:19)
I20160910-20:55:42.774(-5)?     at ScalarField._default [as default](lib/collections/notifications.js:34:48)
I20160910-20:55:42.775(-5)?     at ScalarField.getDefault (packages/jagi:astronomy/lib/modules/fields/field.js:23:14)
I20160910-20:55:42.775(-5)?     at packages/jagi:astronomy/lib/modules/fields/utils/set_defaults.js:12:31
I20160910-20:55:42.776(-5)?     at arrayEach (packages/jagi_astronomy/.2.1.2.prdd1o++os+web.browser+web.cordova/npm/node_modules/lodash/lodash.js:485:11)
I20160910-20:55:42.776(-5)?     at Function.forEach (packages/jagi_astronomy/.2.1.2.prdd1o++os+web.browser+web.cordova/npm/node_modules/lodash/lodash.js:8847:14)
I20160910-20:55:42.777(-5)?     at setDefaults (packages/jagi:astronomy/lib/modules/fields/utils/set_defaults.js:9:5)
I20160910-20:55:42.777(-5)?     at Class (packages/jagi:astronomy/lib/core/class.js:83:7)
I20160910-20:55:42.778(-5)?     at new Class (packages/jagi:astronomy/lib/core/class.js:146:17)
I20160910-21:04:14.463(-5)? Exception in defer callback: Error: Meteor.userId can only be invoked in method calls. Use this.userId in publish functions.
I20160910-21:04:14.466(-5)?     at AccountsServer.userId (packages/accounts-base/accounts_server.js:82:13)
I20160910-21:04:14.467(-5)?     at Object.Meteor.userId (packages/accounts-base/accounts_common.js:246:19)
I20160910-21:04:14.467(-5)?     at ScalarField._default [as default](lib/collections/notifications.js:34:48)
I20160910-21:04:14.467(-5)?     at ScalarField.getDefault (packages/jagi:astronomy/lib/modules/fields/field.js:23:14)
I20160910-21:04:14.468(-5)?     at packages/jagi:astronomy/lib/modules/fields/utils/set_defaults.js:12:31
I20160910-21:04:14.468(-5)?     at arrayEach (packages/jagi_astronomy/.2.1.2.prdd1o++os+web.browser+web.cordova/npm/node_modules/lodash/lodash.js:485:11)
I20160910-21:04:14.469(-5)?     at Function.forEach (/packages/jagi_astronomy/.2.1.2.prdd1o++os+web.browser+web.cordova/npm/node_modules/lodash/lodash.js:8847:14)
I20160910-21:04:14.470(-5)?     at setDefaults (packages/jagi:astronomy/lib/modules/fields/utils/set_defaults.js:9:5)
I20160910-21:04:14.492(-5)?     at Class (packages/jagi:astronomy/lib/core/class.js:83:7)
I20160910-21:04:14.493(-5)?     at new Class (packages/jagi:astronomy/lib/core/class.js:146:17)

**Comment by lukejagodzinski** [comment]
Read this thread: https://github.com/jagi/meteor-astronomy/issues/488

**Comment by rnarayan** [comment]
Thanks. I found out the error and it's not related to Astronomy or the thread #488. The following code was producing the error:

`Meteor.defer( () => sendVerificationEmail(this.userId) )`

When a defer function executes, it may not have this.userId available (the user could have logged out). A google search led me to this thread where I found the answer - https://github.com/meteor/meteor/issues/2221

I changed my code to the following to resolve the error.

```
const userId = this.userId 
Meteor.defer( () => sendVerificationEmail(userId) )
```

Regarding the erroneous message that kind of throws you off in the wrong direction, Glasser explains in the thread - "...we figured when writing the error message that the most likely place you'd call Meteor.userId outside of methods is a publish function."
